src/strands/experimental/bidirectional_streaming/agent/agent.py

```diff
@@ -0,0 +1,167 @@
+from ..utils.debug import log_event, log_flow
+
+logger = logging.getLogger(__name__)
+
+
+class BidirectionalAgent:
+    """Agent for bidirectional streaming conversations.
+    
+    Provides type-safe constructor and session management for real-time 
+    audio/text interaction with concurrent processing capabilities.
+    """
+    
+    def __init__(
+        self,
+        model: BidirectionalModel,
+        tools: Optional[List] = None,
+        system_prompt: Optional[str] = None,
+        messages: Optional[Messages] = None
+    ):
+        """Initialize bidirectional agent with required model and optional configuration.
+        
+        Args:
+            model: BidirectionalModel instance supporting streaming sessions.
+            tools: Optional list of tools available to the model.
+            system_prompt: Optional system prompt for conversations.
+            messages: Optional conversation history to initialize with.
+        """
+        self.model = model
+        self.system_prompt = system_prompt
+        self.messages = messages or []
+        
+        # Initialize tool registry using existing Strands infrastructure
+        self.tool_registry = ToolRegistry()
+        if tools:
+            self.tool_registry.process_tools(tools)
+        self.tool_registry.initialize_tools()
+        
+        # Initialize tool executor for concurrent execution
+        from strands.tools.executors import ConcurrentToolExecutor
+        self.tool_executor = ConcurrentToolExecutor()
+        
+        # Session management
+        self._session = None
+        self._output_queue = asyncio.Queue()
+    
+    async def start_conversation(self) -> None:
+        """Initialize persistent bidirectional session for real-time interaction.
+        
+        Creates provider-specific session and starts concurrent background tasks
+        for model events, tool execution, and session lifecycle management.
+        
+        Raises:
+            ValueError: If conversation already active.
+            ConnectionError: If session creation fails.
+        """
+        if self._session and self._session.active:
+            raise ValueError("Conversation already active. Call end_conversation() first.")
+        
+        log_flow("conversation_start", "initializing session")
+        self._session = await start_bidirectional_connection(self)
+        log_event("conversation_ready")
+    
+    async def send_text(self, text: str) -> None:
+        """Send text input during active session without interrupting model generation.
+        
+        Args:
+            text: Text message to send to the model.
+            
+        Raises:
+            ValueError: If no active session.
+        """
+        self._validate_active_session()
+        log_event("text_sent", length=len(text))
+        await self._session.model_session.send_text_content(text)
+    
+    async def send_audio(self, audio_input: AudioInputEvent) -> None:
```

Comment:
Same as the comment above, we can support __call__ with audio / text blocks for followşng devx